src/internals.jl

```diff
@@ -0,0 +1,200 @@
+module Internals
+
+using ChunkSplitters: Split, Consecutive, RoundRobin
+import ChunkSplitters: chunk_indices, chunk, is_chunkable
+import Base: iterate, length, eltype, enumerate, firstindex, lastindex, getindex, eachindex
+
+abstract type Constraint end
+struct FixedCount <: Constraint end
+struct FixedSize <: Constraint end
+
+abstract type ReturnType end
+struct ReturnIndices <: ReturnType end
+struct ReturnViews <: ReturnType end
```

Comment:
> I implemented a single iterator ChunksIterator that has a type parameter ReturnIndices or ReturnViews to indicate what will be eventually returned.

A bit of a drive-by comment, but unless you expect the number of `ReturnType`s to expand, isn't it cleaner to have two types called e.g. `IndexChunks` and `Chunks` (or `DataChunks` or `ViewChunks`)?

Reply:
Keeping them as one type saved me a little bit of code duplication. But, yeah, I think it's cleaner and worth it. (Done)